clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -419,6 +483,25 @@ public void run() {
         return Collections.unmodifiableMap(childrenSensors);
     }
 
+    public MetricName metricInstance(MetricNameTemplate template, String... keyValue) {
```

Comment:
With this new method, should we mark all existing Metrics.metricName() deprecated in favor of the new methods?